pkg/zson/zson.go

```diff
@@ -11,32 +11,15 @@ type Reader interface {
 	Read() (*Record, error)
 }
 
-type WriteCloser interface {
+type Writer interface {
 	Write(*Record) error
-	Close() error
 }
 
-// Writer is a simple, embeddable object for zson writers who don't need
-// to do anything special at close.  Instead this object will just call the
-// Close method on the wrapped io.WriteCloser.
-type Writer struct {
-	io.WriteCloser
+type WriteFlusher interface {
```

Comment:
By analogy to bufio.Writer, maybe just call this Writer? Doing this also makes the WriteCloser name more meaningful (as things currently stand, it's a bit odd that WriteFlusher explicitly calls out "Flusher" while WriteCloser doesn't).

Reply:
good point

Reply:
However, it's nice to be able to use the reader/writers externally without having to necessarily implement an empty Flush method (or previously a Close method).  zsio.Lookup has a wrapper so it can generically return WriteFlusher but there will be times we you just want to use a Writer.

Reply:
I think I'll leave it as is for now and we can revisit again if this is frustrating...

Reply:
Maybe WriteCloser has the wrong name (it's a struct rather than an interface)

Reply:
Ok, how about this
* move zsio.Writer to pkg/zsio/zson/ (this has a name conflit with pkg/zson but should be manageable)
* move zson.WriteCloser to zsio.Writer
* have zsio.LookupWriter take an io.Closer and return a zsio.Writer

Reply:
That sounds good 👍